docs/datacontract.md

```diff
@@ -102,3 +102,259 @@ The top-level key within a FileDefinition serves as the FileDefinition name. Thi
 | params   | Dictionary of task parameters                 | N        |
 
 
+#### Supported Tasks
```

Comment:
This is just my .02 but I would prefer markdown tables to HTML markup :) Both formats have their pros and cons. I just feel that markdown is more concise. IDE plugins may help with the experience as well. 

If the team does decide to go with HTML here, I would suggest that we standardize on that and update other sections of the documentation.

Reply:
do you know how to get the json formatting in mark down tables? everything I saw said you had to go to html, was only planning on doing it for this table.  maybe vscode has a good plugin to help?

Reply:
discussed and will clarify in contributing that md is preferred, but html can be used for complex tables in documentation